baseline/client/tests/spatial-orientation.test.js

```diff
@@ -1,7 +1,70 @@
+    const rect = icirc.getBoundingClientRect();
+    icirc.dispatchEvent(new MouseEvent("click", {
+        clientX: +x + icirc.width/2 + rect.left,
+        clientY: -y + icirc.height/2 + rect.top,
+    }));
+};
 
 describe("spatial-orientation", () => {
-    it.todo("results should have at least one result marked isRelevant");
+    it("results should have at least one result marked isRelevant", () => {
+        jest.useFakeTimers("legacy");
+        const timeline = (new SpatialOrientation(1)).getTimeline();
+        jsPsych.init({timeline: timeline});
+        timeline.forEach(() => {
+            pressKey(" ");
+            clickIcirc(document.getElementById("jspsych-spatial-orientation-icirc"), 0, 0);
+            jest.runAllTimers();
+        });
+        const progress = jsPsych.progress();
+        expect(progress.current_trial_global).toBe(progress.total_trials);
+        expect(jsPsych.data.get().filter({isRelevant: true}).values().length).toBeGreaterThan(0);
+    });
+
+    it("completes test trials with timedout and skipped if 5 minutes have elapsed", () => {
+        const advanceDateNow = (dateNow, ms) => () => dateNow() + ms;
+        const originalDateNow = Date.now();  // original Starwalker
+        jest.useFakeTimers("legacy");
+        // skip timeline to test block
+        const timeline = (new SpatialOrientation(1)).getTimeline();
+        jsPsych.init({timeline: timeline});
+        for (const trial of timeline) {
+            if (trial.data?.isRelevant === true) {
+                break;
+            } else if (trial.type === "html-keyboard-response") {
+                pressKey(" ");
+            } else if (trial.type === "spatial-orientation") {
+                clickIcirc(document.getElementById("jspsych-spatial-orientation-icirc"), 0, 0);
+                jest.runAllTimers();
+            }
+        }
+        // expect test trials to NOT be completed at their start
+        expect(jsPsych.data.get().filter({isRelevant: true}).values().length).toBe(0);
+        // expect test trials to still NOT be completed after 4 minutes and 55 seconds
+        {
+            const ms = 4*60*1000 + 55*1000;  // 4 minutes and 55 seconds
+            Date.now = advanceDateNow(Date.now, ms);
+            jest.advanceTimersByTime(ms);
+        }
+        expect(jsPsych.data.get().filter({isRelevant: true}).values().length).toBe(0);
+        // expect test trials to be completed after 5 minutes and 5 seconds
+        {
+            const ms = 10*1000;  // 10 seconds more
+            Date.now = advanceDateNow(Date.now, ms);
+            jest.advanceTimersByTime(ms);
+        }
+        const relevant = jsPsych.data.get().filter({isRelevant: true}).values();
+        expect(relevant.length).toBe(12);
+        expect(relevant[0].completionReason).toBe("timedout");
+        relevant.slice(1).forEach(t => {
+             expect(t.completionReason).toBe("skipped");
+        });
+        Date.now = originalDateNow;
```

Comment:
If something above fails this won't get called, leaving Date.now set to your mock function. Please use [beforeEach/afterEach](https://jestjs.io/docs/setup-teardown) to make sure that things get cleaned up and Date.now is set to the original.